packages/liveblocks-core/src/room.ts

```diff
@@ -3057,98 +2655,461 @@ export function createRoom<
+    } else {
+      return context.unacknowledgedOps.size === 0
+        ? "synchronized"
+        : "synchronizing";
+    }
+  }
+
+  /**
+   * Storage status is a computed value based other internal states so we need to keep a reference to the previous computed value to avoid triggering events when it does not change
+   * This is far from ideal because we need to call this function whenever we update our internal states.
+   *
+   * TODO: Encapsulate our internal state differently to make sure this event is triggered whenever necessary.
+   * Currently okay because we only have 4 callers and shielded by tests.
+   */
+  let _lastStorageStatus = getStorageStatus();
+  function notifyStorageStatus() {
+    const storageStatus = getStorageStatus();
+    if (_lastStorageStatus !== storageStatus) {
+      _lastStorageStatus = storageStatus;
+      eventHub.storageStatus.notify(storageStatus);
+    }
+  }
+
+  function isPresenceReady() {
+    return self.current !== null;
+  }
+
+  async function waitUntilPresenceReady(): Promise<void> {
+    while (!isPresenceReady()) {
+      const { promise, resolve } = Promise_withResolvers();
+
+      const unsub1 = events.self.subscribeOnce(resolve);
+      const unsub2 = events.status.subscribeOnce(resolve);
+      // Return whenever one of these returns, whichever is first
+      await promise;
+      unsub1();
+      unsub2();
+    }
+  }
+
+  function isStorageReady() {
+    return getStorageSnapshot() !== null;
+  }
+
+  async function waitUntilStorageReady(): Promise<void> {
+    while (!isStorageReady()) {
+      // Trigger a load of Storage and wait until it finished
+      await getStorage();
+    }
+  }
+
+  // Derived cached state for use in DevTools
+  const others_forDevTools = new DerivedRef(context.others, (others) =>
+    others.map((other, index) => userToTreeNode(`Other ${index}`, other))
+  );
+
+  const events = {
+    status: eventHub.status.observable,
+    lostConnection: eventHub.lostConnection.observable,
+
+    customEvent: eventHub.customEvent.observable,
+    others: eventHub.others.observable,
+    self: eventHub.self.observable,
+    myPresence: eventHub.myPresence.observable,
+    error: eventHub.error.observable,
+    /** @deprecated */
+    storage: eventHub.storageBatch.observable,
+    storageBatch: eventHub.storageBatch.observable,
+    history: eventHub.history.observable,
+    storageDidLoad: eventHub.storageDidLoad.observable,
+    storageStatus: eventHub.storageStatus.observable,
+    ydoc: eventHub.ydoc.observable,
+
+    comments: eventHub.comments.observable,
+  };
+
+  async function fetchCommentsApi(
+    endpoint: string,
+    params?: QueryParams,
+    options?: RequestInit
+  ): Promise<Response> {
+    // TODO: Use the right scope
+    const authValue = await delegates.authenticate();
+
+    return fetchClientApi(config.roomId, endpoint, authValue, options, params);
+  }
+
+  async function fetchCommentsJson<T>(
```

Comment:
Looks like this is a new function, right? Did you intend to include it in this PR? I'm fine with it, but just double-checking since you mentioned it was only moving code around.

Reply:
🦅👁️ `createCommentsApi` had it as `fetchJson` so I renamed it to fit in `createRoom`, but its implementation hasn't changed.